admin_2017.rb

```diff
@@ -20,110 +20,138 @@ def source_dir
+RUBY
 
 ####################################
-# CONTROLLERS
+# COPY CONFIG
 ####################################
 
-copy_file 'admin.haml', 'app/views/layouts/admin.haml'
-
-file 'app/controllers/admin/base_controller.rb', <<~'RUBY'
-  module Admin
-    class BaseController < ApplicationController
-      include AdminAuthenticable
-      # include AdminSource
-      before_action :authenticate!
-      # before_action :ensure_admin_domain
+copy_file 'config/initializers/carrierwave.rb', 'config/initializers/carrierwave.rb'
 
-      layout 'admin'
-    end
-  end
+environment <<~'RUBY'
+  # specify locales avaialable to rails-i18n gem
+  config.i18n.available_locales = [:en, :es]
 RUBY
+directory 'config/locales', 'config/locales'
 
-copy_file 'sessions_controller.rb', 'app/controllers/admin/sessions_controller.rb'
-copy_file 'new.haml', 'app/views/admin/sessions/new.haml'
+copy_file 'config/webpack/environment.js', 'config/webpack/environment.js', force: true
 
-copy_file 'dashboard_controller.rb', 'app/controllers/admin/dashboard_controller.rb'
-copy_file 'dashboard.haml', 'app/views/admin/dashboard/show.haml'
 
 ####################################
-# ASSETS
+# COPY STUB MIGRATIONS
 ####################################
+directory 'db/migrate', 'db/migrate', recursive: true, verbose: true # TODO make other migrations not hard coded
+generate 'devise AdminUser'
+directory 'spec/fabricators', 'spec/fabricators', recursive: true, verbose: true
 
-# insert_into_file 'config/webpack/shared.js', after: '  plugins: [' do
-#   "new webpack.ProvidePlugin({jQuery: 'jquery'}),"
-# end
+####################################
+# COPY APP CODE HIERARCHY
+####################################
+directory 'app', 'app', recursive: true, verbose: true
+
+####################################
+# COPY LIB HIERARCHY
+####################################
+directory 'lib', 'lib', recursive: true, verbose: true
 
-run 'yarn add jquery jquery-ujs turbolinks bootstrap-sass'
-
-file 'app/javascript/admin/admin.scss', <<~'SCSS'
-  $icon-font-path: '~bootstrap-sass/assets/fonts/bootstrap/';
-  @import '~bootstrap-sass/assets/stylesheets/_bootstrap.scss';
-SCSS
-
-file 'app/javascript/packs/admin.js', <<~'JAVASCRIPT'
-  import 'admin/admin.scss'
-  import * as $ from 'jquery'
-  import 'jquery-ujs'
-  import Turbolinks from 'turbolinks'
-
-  // TODO: fix these
-  // import 'bootstrap/js/dropdown'
-  // import 'bootstrap/js/modal'
-  // import 'bootstrap/js/tooltip'
-  // import 'bootstrap/js/popover'
-  // import 'bootstrap/js/alert'
-  // import 'bootstrap/js/transition'
-  // import 'bootstrap/js/carousel'
-
-  Turbolinks.start()
-
-  document.addEventListener('DOMContentLoaded', () => {
-    // once ever
-  })
-
-  document.addEventListener('turbolinks:load', () => {
-    // every page
-  })
-JAVASCRIPT
+####################################
+# JS
+####################################
+run 'yarn add bootstrap@3 bootstrap-sass jquery jquery-ujs turbolinks'
```

Comment:
are we using turbolinks in either the app or admin?

Reply:
I copied over vitacoco's javascript and that used turbolinks.  I believe admin has a Turbolinks.start, although I don't know why an admin javascript needs it thinking about it now.  Could remove it, if it seems unnecessary.